pkg/kubectl/cmd/drain.go

```diff
@@ -282,16 +283,28 @@ func (o *DrainOptions) getController(sr *api.SerializedReference) (interface{},
 	return nil, fmt.Errorf("Unknown controller kind %q", sr.Reference.Kind)
 }
 
+func newSerializedControllerRef(pod api.Pod) *api.SerializedReference {
+	return &api.SerializedReference{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "v1",
+			Kind:       "SerializedReference",
+		},
+		Reference: api.ObjectReference{ //currently only set based on first controller
+			APIVersion: pod.GetOwnerReferences()[0].APIVersion,
+			Kind:       pod.GetOwnerReferences()[0].Kind,
+			Name:       pod.GetOwnerReferences()[0].Name,
+			Namespace:  pod.GetObjectMeta().GetNamespace(),
+		},
+	}
+}
+
 func (o *DrainOptions) getPodCreator(pod api.Pod) (*api.SerializedReference, error) {
-	creatorRef, found := pod.ObjectMeta.Annotations[api.CreatedByAnnotation]
-	if !found {
+	if controller.GetControllerOf(pod.GetObjectMeta()) == nil {
 		return nil, nil
 	}
 	// Now verify that the specified creator actually exists.
-	sr := &api.SerializedReference{}
-	if err := runtime.DecodeInto(o.Factory.Decoder(true), []byte(creatorRef), sr); err != nil {
-		return nil, err
-	}
+	sr := newSerializedControllerRef(pod)
```

Comment:
The `SerializedReference` type was only used here because it's the type of object that was stored in the `CreatedByAnnotation`. Instead of translating the ControllerRef back into a `SerializedReference`, I suggest to modify `getController()` and all callers of `getPodCreator()` to work based on a value of type `*metav1.OwnerReference`. Then you can directly use the value returned by `controller.GetControllerOf()` (if it's not `nil`).